Tests/Model/SPIClientTests.m

```diff
@@ -15,24 +15,86 @@ @interface SPIClientTests : XCTestCase
 
 @implementation SPIClientTests
 
-- (void)testSetPosId {
-    NSString *posId = @"test";
+- (void)testSetPosId_OnValidCharacters_IsSet {
+    // arrange
+    static NSString *posId = @"test";
+    static NSString *regexItemsForPosId = @"^[a-zA-Z0-9 ]*$";
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:regexItemsForPosId options:NSRegularExpressionCaseInsensitive error:nil];
```

Comment:
Are you testing that `NSRegularExpression` works? That's redundant, you didn't write it.

Reply:
Removed tests

Reply:
The one I commented about is still there...

Reply:
Removed